src/main/java/dk/cngroup/lentils/controller/ClientContactController.java

```diff
@@ -0,0 +1,51 @@
+package dk.cngroup.lentils.controller;
+
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.security.CustomUserDetails;
+import dk.cngroup.lentils.service.ContactService;
+import dk.cngroup.lentils.service.CypherService;
+import dk.cngroup.lentils.service.ScoreService;
+import dk.cngroup.lentils.service.StatusService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/contact")
+public class ClientContactController {
+    private static final String VIEW_CONTACT = "client/contact";
+    private final ContactService contactService;
+    private final ScoreService scoreService;
+    private final CypherService cypherService;
+    private final StatusService statusService;
+
+    @Autowired
+    public ClientContactController(final ContactService contactService,
+                                   final ScoreService scoreService,
+                                   final CypherService cypherService,
+                                   final StatusService statusService) {
+        this.contactService = contactService;
+        this.scoreService = scoreService;
+        this.cypherService = cypherService;
+        this.statusService = statusService;
+    }
+
+    @GetMapping
+    public String contact(@AuthenticationPrincipal final CustomUserDetails user,
+                          final Model model) {
+        List<Cypher> cyphers = cypherService.getAllCyphersOrderByStageAsc();
+        if (statusService.isStatusInDbByCypherAndTeam(cyphers.get(0), user.getTeam())) {
```

Comment:
Tady jsem uz v nejakem jinek kontroleru nasel chybku a hlasil issue.
Nekontroluje se, jestli tam ten multu prvek je :) pak to spadne na outOfBounds vyjimce.
Takze v tom ifu bych jeste kontroloval, jestli je ten list prazdny.

pseudokod:
if (ifNotEmpty and statusService....)

Reply:
opraveno, snad jeto OK - nedala jsem to do 1 ifu, protoze si nejsu jista, jestli po vyhodnoceni prvni podminky v && jako false uz ke druhe nejde - pokud by kontroloval i tu druhou, tak by to spadlo stejne.

Reply:
opraveno znovu